lib/DriverDisplay/DriverDisplay.cpp

```diff
@@ -76,652 +104,642 @@ float pvLast = -1;
-    }
-
-    xSemaphoreGive(spi_mutex);
-    // CRITICAL SECTION SPI: end
-
-    //tft.setFont(&FreeSans9pt7b);
-    driver_display_demo_screen();
-    delay(1000);
-    write_driver_info("ready.", INFO_TYPE::INFO);
-    indicator_set_and_blink(INDICATOR::OFF, false);
-    write_speed(0);
-    write_acceleration(0);
-    _arrow_increase(bgColor);
-    _arrow_decrease(bgColor);
-    _light1(false);
-    _light2(false);
-    write_bat(0.0);
-    write_pv(0.0);
-    write_motor(0.0);
-
-    vTaskDelay(500 / portTICK_PERIOD_MS);
+  tft.begin();
+  try {
+    uint8_t x = tft.readcommand8(ILI9341_RDMODE);
+    printf("Display Power Mode: 0x%x\n", x);
+    x = tft.readcommand8(ILI9341_RDMADCTL);
+    printf("MADCTL Mode:        0x%x\n", x);
+    x = tft.readcommand8(ILI9341_RDPIXFMT);
+    printf("Pixel Format:       0x%x\n", x);
+    x = tft.readcommand8(ILI9341_RDIMGFMT);
+    printf("Image Format:       0x%x\n", x);
+    x = tft.readcommand8(ILI9341_RDSELFDIAG);
+    printf("Self Diagnostic:    0x%x\n", x);
+    infoFrameSizeX = tft.width();
+    speedFrameX = (tft.width() - speedFrameSizeX) / 2;
+    printf("[Display ILI9341] Initialize successfully with screen %d x %d.\n",
+           tft.height(), tft.width());
+  } catch (__exception ex) {
+    printf("[Display ILI9341] Unable to initialize screen.\n");
+  }
+
+  xSemaphoreGive(spi_mutex);
+  // CRITICAL SECTION SPI: end
+
+  // tft.setFont(&FreeSans9pt7b);
+  driver_display_demo_screen();
+  delay(1000);
+  write_driver_info("ready.", INFO_TYPE::INFO);
+  indicator_set_and_blink(INDICATOR::OFF, false);
+  write_speed(0);
+  write_acceleration(0);
+  _arrow_increase(bgColor);
+  _arrow_decrease(bgColor);
+  _light1(false);
+  _light2(false);
+  write_bat(0.0);
+  write_pv(0.0);
+  write_motor(0.0);
+
+  printf("[v] Driver Display inited: SPI_CS_TFT=%d, SPI_DC=%d, SPI_MOSI=%d, "
+         "SPI_CLK=%d, SPI_RST=%d, SPI_MISO=%d.\n",
+         SPI_CS_TFT, SPI_DC, SPI_MOSI, SPI_CLK, SPI_RST, SPI_MISO);
+  vTaskDelay(500 / portTICK_PERIOD_MS);
 }
 
 // -------------
 // FreeRTOS TASK
 // -------------
-void driver_display_task(void *pvParameter)
-{
-    // polling loop
-    while (1)
-    {
-        // long start = micros();
-
-        // // CRITICAL SECTION SPI: start
-        // xSemaphoreTake(spi_mutex, portMAX_DELAY);
-        // xSemaphoreGive(spi_mutex);
-        // // CRITICAL SECTION SPI: end
-
-        if (lifeSignCounter > 10)
-        {
-            lifeSign();
-            lifeSignCounter = 0;
-        }
-        lifeSignCounter++;
-
-        //printf("time elapsed: %ld\n", micros() - start);
-
-        // sleep for 1s
-        vTaskDelay(100 / portTICK_PERIOD_MS);
+void driver_display_task(void *pvParameter) {
+  // polling loop
+  while (1) {
+    // long start = micros();
+
+    // // CRITICAL SECTION SPI: start
+    // xSemaphoreTake(spi_mutex, portMAX_DELAY);
+    // xSemaphoreGive(spi_mutex);
+    // // CRITICAL SECTION SPI: end
```

Comment:
Why are you keeping commented out code in?

Reply:
removed